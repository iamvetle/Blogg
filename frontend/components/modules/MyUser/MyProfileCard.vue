<template>
    <div id="personal-user-card" class="mx-auto mb-8 px-8 py-4 rounded-lg text-onSecondaryContainer bg-secondaryContainer shadow-md w-full">
        <div class="mb-4 w-full flex flex-wrap min-w-0 items-center text-onPrimaryContainer">
            <img :src="userProp.avatar" class="w-20 h-auto text-onPrimaryContainer mr-2">
            <div class="min-w-0">
                <h2 id="card-username" class="text-4xl text-onPrimaryContainer whitespace-normal break-words">{{ userProp.username }}</h2>
                <span>
                    <p id="card-name" class="text-onPrimaryContainer ">{{ full_name }}</p>
                </span>
            </div>
        </div>
        <p class="text-onPrimaryContainer">Antall f√∏lgere: X</p>
        <p class="text-onPrimaryContainer">Antall innlegg: X</p>
    </div>
</template>

<script setup lang="ts">
//import profile_picture from '~/assets/account-circle-line.svg'

const { userProp } = defineProps(["userProp"])

const full_name = computed(() => {
    return userProp.first_name + " " + userProp.last_name
})

</script>

<style scoped></style>