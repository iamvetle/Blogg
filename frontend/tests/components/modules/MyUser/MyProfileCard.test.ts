import MyProfileCard from '~/components/modules/MyUser/MyProfileCard.vue';
import { VueWrapper, shallowMount } from '@vue/test-utils';

describe('testing myprofilecard', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(MyProfileCard, {
      global: {
        mocks: {},
        stubs: {},
      },
      slots: {
        username:"testusername",
        "full-name":"test_fulltnavn",
        "amount-of-followers":"243",
      },
      props: {
        profilePictureProp: "~/src/something.jpg"
      }
    });
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
    }
  });

  test("props are being passed correctly", () => {
    const ell = wrapper.get("[data-test='imgprop']")

    expect(ell.attributes("src")).toBe("~/src/something.jpg")

    expect(wrapper.text()).toMatchSnapshot()

  })
});