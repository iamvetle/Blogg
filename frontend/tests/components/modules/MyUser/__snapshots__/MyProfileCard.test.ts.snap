// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing myprofilecard > props are being passed correctly 1`] = `"testusernametest_fulltnavn243Antall lagrede innlegg"`;
